File: spec/ruby/run_benchmark_process_xor.rb
#!/usr/bin/env ruby

require 'benchmark_process_xor'
require 'benchmark'

r = nil
max = nil

t1 = Benchmark::measure {
  r = BenchmarkProcessXor::from_file("#{ENV['DATA_DIR']}/benchmark_process_xor.dat")
}

t2 = Benchmark::measure {
  max = r.chunks.map { |chunk| chunk.body.numbers.max }.max
}

puts "max = #{max}"
p t1
p t2

File: spec/python/run_benchmark_process_xor.py
#!/usr/bin/env python

import os
from timeit import default_timer as timer
from benchmark_process_xor import BenchmarkProcessXor

t1 = timer()

r = BenchmarkProcessXor.from_file("%s/benchmark_process_xor.dat" % os.environ['DATA_DIR'])

t2 = timer()

sum = 0
for chunk in r.chunks:
    for n in chunk.body.numbers:
        sum += n

t3 = timer()

print("parsing: %s seconds" % (t2-t1,))
print("processing: %s seconds" % (t3-t2,))
